engine/crates/engine-v2/schema/src/conversion.rs

```diff
@@ -281,3 +299,9 @@ from_id_newtypes! {
     SubgraphId => SubgraphId,
     UnionId => UnionId,
 }
+
+impl DataSourceId {
+    fn from_subgraph_id(id: federated_graph::SubgraphId) -> Self {
+        Self(NonZeroU32::new(id.0 as u32 + 1).unwrap())
```

Comment:
Not sure I'm _against_ these `NonZeroU32`s - but they do seem like they're increasing our risk of off by one errors significantly...

Reply:
Very fair point, moved all the IDs into a sub-module so the only way to create them is through `From<usize>`. As I'm using `Option<ObjectId>` in response objects it felt like a waste of space to allocate additional bytes.